server/galaxyls/services/xsd/types.py

```diff
@@ -149,3 +152,16 @@ def _get_path_from_stack(self, node_stack: List[str]) -> str:
         if node_stack[0] == self.root.name:
             node_stack[0] = "."
         return "/".join(node_stack)
+
+    def _build_expand_element(self) -> XsdNode:
+        """Creates a XsdNode representing the <expand> element and it's attributes.
+        The <expand> element is a special element that is not defined in the XSD schema but allows to use macros in the tool.
+
+        Returns:
+            XsdNode: The node with information about the special <expand> element.
+        """
+        expand_node = XsdNode("expand", None)
+        attr_name = "macro"
+        attr = XsdAttribute(attr_name, None, type_name=None, is_required=True)
```

Comment:
could we actually maintain a list of macro names and offer them as selection here?

Reply:
Sure! We can use a list of all the declared macro names and tokens for use when auto-completing.

Reply:
Only macros here, the tokens go into command or help section (most of the times).